src/differenceInBusinessDays/index.ts

```diff
@@ -85,14 +87,59 @@ export default function differenceInBusinessDays(
   const weeks = toInteger(calendarDifference / 7)
 
   let result = weeks * businessDays.length
-  dateRight = addDays(dateRight, weeks * 7)
+  let newDateRight = addDays(dateRight, weeks * 7)
 
-  // the loop below will run at most 6 times to account for the remaining days that don't makeup a full week
-  while (!isSameDay(dateLeft, dateRight)) {
+  // the loop below will run at most 6 times to account for the remaining days that don't make up a full week
+  while (!isSameDay(dateLeft, newDateRight)) {
     // sign is used to account for both negative and positive differences
-    result += isHoliday(dateRight) ? 0 : sign
-    dateRight = addDays(dateRight, sign)
+    result += isHoliday(newDateRight) ? 0 : sign
+    newDateRight = addDays(newDateRight, sign)
   }
 
-  return result === 0 ? 0 : result
+  // handle exceptions
+  let exceptionCount = 0
+  if (options.exceptions) {
+    const exceptionDates = Object.keys(options.exceptions)
+    exceptionDates.forEach((e) => {
+      const date = new Date(e)
+      if (!isValid(date)) return
+      // if date is within the left and right dates
+      if (isBefore(date, dateLeft) && isAfter(date, dateRight)) {
```

Comment:
thought: I wonder if we could extract/share this filtering logic of exceptions?

Reply:
The differenceInBusinessDays function has some extra edge cases that make the logic difficult to share. For now I think I'll go with these functions as is and try using them in GQL service.